pkg/controller/machine.go

```diff
@@ -404,13 +404,24 @@ func (c *Controller) syncHandler(key string) error {
 					return fmt.Errorf("failed to create bootstrap kubeconfig: %v", err)
 				}
 
-				data, err := userdataProvider.UserData(machine.Spec, kubeconfig, prov, c.clusterDNSIPs)
+				//Need the kubeconfig to extract the CACert for the userdata rendering
+				cm, err := c.kubeconfigProvider.GetKubeconfig()
 				if err != nil {
-					return fmt.Errorf("failed get userdata: %v", err)
+					return fmt.Errorf("failed to get kubeconfig: '%v'", err)
+				}
+				if len(cm.Clusters) != 1 {
+					return fmt.Errorf("kubeconfig does not contain exactly one cluster, can not extract cacert...")
+				}
+				var userData string
+				for _, clusterConfig := range cm.Clusters {
```

Comment:
Why iterating over all clusters?
If there's more than 1 cluster, we should throw an error. As we cannot determine which cluster to use

Reply:
Because its a map and we cant know the right key in advance